go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/celestiaorg/celestia-app/v3
 
 go 1.23.6
 
+toolchain go1.24.1
```

Comment:
This makes me think we should bump the Go directive and the Go version recommended in the README